apiendpoint/api_endpoint.go

```diff
@@ -94,6 +95,9 @@ type MountOpts struct {
 	// MiddlewareStack is a stack of middleware that will be mounted in front of
 	// the API endpoint handler. If not specified, no middleware will be used.
 	MiddlewareStack *apimiddleware.MiddlewareStack
+	// Validator is the validator to use for this endpoint. If not specified,
+	// the default validator will be used.
+	Validator *validator.Validate
```

Comment:
The only thing is that it seems like a bit of a smell to have this configured on a per-endpoint basis. Very likely you'd want one custom validator that you're going to be using application-wide, and having to inject that into every single endpoint seems like a large-ish chore and would add quite a lot of visual noise to the definitions.

Reply:
Yes, the issue is there’s no global struct to keep it on. I will say in practice it turned out to not be bad because I’m reusing the same mount opts struct for all endpoints in each group (authenticated and not).